src/main/java/minijava/parser/Parser.java

```diff
@@ -0,0 +1,450 @@
+  private void parseTypeIdentFieldOrMethod() {
+    parseType();
+    expectTokenAndConsume(Terminal.IDENT);
+    parseFieldOrMethod();
+  }
+
+  /** FieldOrMethod -> ; | Method */
+  private void parseFieldOrMethod() {
+    if (isCurrentTokenTypeOf(Terminal.SEMICOLON)) {
+      expectTokenAndConsume(Terminal.SEMICOLON);
+    } else {
+      parseMethod();
+    }
+  }
+
+  /** Method -> ( Parameters? ) Block */
+  private void parseMethod() {
+    expectTokenAndConsume(Terminal.LPAREN);
+    if (isCurrentTokenNotTypeOf(Terminal.RPAREN)) {
+      parseParameters();
+    }
+    expectTokenAndConsume(Terminal.RPAREN);
+    parseBlock();
+  }
+
+  /** Parameters -> Parameter | Parameter , Parameters */
+  private void parseParameters() {
+    parseParameter();
+    if (isCurrentTokenTypeOf(Terminal.COMMA)) {
+      parseParameters();
+    }
+  }
+
+  /** Parameter -> Type IDENT */
+  private void parseParameter() {
+    parseType();
+    expectTokenAndConsume(Terminal.IDENT);
+  }
+
+  /** Type -> BasicType ([])* */
+  private void parseType() {
+    parseBasicType();
+    while (isCurrentTokenTypeOf(Terminal.LBRACKET) && isCurrentTokenNotTypeOf(Terminal.EOF)) {
+      expectTokenAndConsume(Terminal.LBRACKET);
+      expectTokenAndConsume(Terminal.RBRACKET);
+    }
+  }
+
+  /** BasicType -> int | boolean | void | IDENT */
+  private void parseBasicType() {
+    switch (currentToken.terminal) {
+      case INT:
+        expectTokenAndConsume(Terminal.INT);
+        break;
+      case BOOLEAN:
+        expectTokenAndConsume(Terminal.BOOLEAN);
+        break;
+      case VOID:
+        expectTokenAndConsume(Terminal.VOID);
+        break;
+      case IDENT:
+        expectTokenAndConsume(Terminal.IDENT);
+        break;
+    }
+  }
+
+  /**
+   * Statement -> Block | EmptyStatement | IfStatement | ExpressionStatement | WhileStatement |
+   * ReturnStatement
+   */
+  private void parseStatement() {
+    switch (currentToken.terminal) {
+      case LCURLY:
+        parseBlock();
+        break;
+      case SEMICOLON:
+        parseEmptyStatement();
+        break;
+      case IF:
+        parseIfStatement();
+        break;
+      case WHILE:
+        parseWhileStatement();
+        break;
+      case RETURN:
+        parseReturnStatement();
+        break;
+      default:
+        parseExpressionStatement();
+        break;
+    }
+  }
+
+  /** Block -> { BlockStatement* } */
+  private void parseBlock() {
+    expectTokenAndConsume(Terminal.LCURLY);
+    while (isCurrentTokenNotTypeOf(Terminal.RCURLY) && isCurrentTokenNotTypeOf(Terminal.EOF)) {
+      parseBlockStatement();
+    }
+    expectTokenAndConsume(Terminal.RCURLY);
+  }
+
+  /** BlockStatement -> Statement | LocalVariableDeclarationStatement */
+  private void parseBlockStatement() {
+    switch (currentToken.terminal) {
+      case INT:
+      case BOOLEAN:
+      case VOID:
+      case IDENT:
+        parseLocalVariableDeclarationStatement();
```

Comment:
The problem is IDENT, isn't it? Your statements need to be parsed as Statement and not as LocalVariableDeclarationStatement, because it is not a declaration.